src/d/actor/d_a_player_swim.inc

```diff
@@ -11,85 +11,613 @@
+
+        offNoResetFlg0(daPyFlg0_EQUIP_HEAVY_BOOTS);
+        offNoResetFlg1(daPyFlg1_EQUIP_DRAGON_SHIELD);
+
+        endFlameDamageEmitter();
+
+        offNoResetFlg0(daPyFlg0_UNK100);
+
+        mMaxNormalSpeed = l_HIO_swim_c0.m.field_0xC;
+        mVelocity *= 0.75f;
+
+        if (mVelocity >= mMaxNormalSpeed) {
+          mVelocity = mMaxNormalSpeed;
+        }
+
+        m35C4 = 0.0f;
+
+        setSwimTimerStartStop();
+        
+        if (checkModeFlg(2)) {
+            current.pos.y += l_HIO_swim_c0.m.field_0x24;
+            speed.y = mOldSpeed.y * l_HIO_swim_c0.m.field_0x10;
+
+            if (speed.y < l_HIO_swim_c0.m.field_0x14) {
+                speed.y = l_HIO_swim_c0.m.field_0x14;
+            }
+            else if (speed.y > 0.0f) {
+                speed.y = 0.0f;
+            }
+
+            float oldSpeedY = mOldSpeed.y * mOldSpeed.y * 0.0004f + 0.2f;
+
+            if (oldSpeedY > 1.0f) {
+                oldSpeedY = 1.0f;
+            }
+
+            local_18.x = current.pos.x;
+            local_18.y = m35D0;
+            local_18.z = current.pos.z;
+            fopKyM_createWpillar(&local_18, 1.0f, oldSpeedY, 0);
+            seStartOnlyReverb(JA_SE_LK_INTO_WATER);
+            res = procSwimWait_init(0);
+        }
+        else {
+            res = procSwimUp_init(0);
+        }
+    }
+    else {
+        res = 0;
+    }
+    return res;
 }
 
 /* 8013D1B0-8013D2AC       .text changeSwimUpProc__9daPy_lk_cFv */
-void daPy_lk_c::changeSwimUpProc() {
-    /* Nonmatching */
+BOOL daPy_lk_c::changeSwimUpProc() {
+    if (speed.y < l_HIO_swim_c0.m.field_0x68) {
+        speed.y += l_HIO_swim_c0.m.field_0x64;
+        if (speed.y > l_HIO_swim_c0.m.field_0x68) {
+            speed.y = l_HIO_swim_c0.m.field_0x68;
+        }
+        if (speed.y > 0.0f && mEquipItem != 0x100) { // No value for 0x100 in `ItemTable`?
+            deleteEquipItem(1);
+        }
+    }
+    if (m35D0 - current.pos.y < 40.1f + l_HIO_swim_c0.m.field_0x28 && speed.y >= 0.0f) {
+        if (mCurProc == 0xb2) {
+            onNoResetFlg0(daPyFlg0_UNK100);
+            current.pos.y = m35D0;
+            return TRUE;
+        }
+        return procSwimUp_init(1);
+    }
+    return FALSE;
 }
 
 /* 8013D2AC-8013D314       .text swimOutAfter__9daPy_lk_cFi */
-void daPy_lk_c::swimOutAfter(int) {
-    /* Nonmatching */
+void daPy_lk_c::swimOutAfter(int param_1) {
+    onNoResetFlg0(daPyFlg0_UNK100);
+    m35C4 = 0.0f;
+    if (param_1) {
+        seStartOnlyReverb(JA_SE_LK_OUTOF_WATER);
+    }
+    dComIfGp_clearItemTimeCount();
+    dComIfGp_setItemTimeMax(0);
+    resetPriTextureAnime();
 }
 
 /* 8013D314-8013D354       .text checkSwimFallCheck__9daPy_lk_cFv */
 BOOL daPy_lk_c::checkSwimFallCheck() {
-    /* Nonmatching */
+    if (!checkNoResetFlg0(daPyFlg0_UNK80) || current.pos.y > m35D0 + 30.1f && !(dPa_control_c::mStatus & 1)) { // Fakematch? (dPa_control_c::mStatus & 1) seems like an inline
```

Comment:
```suggestion
    if (!checkNoResetFlg0(daPyFlg0_UNK80) || current.pos.y > m35D0 + 30.1f && !dPa_control_c::isStatus(1)) {
```
You can "Find all references" (alt+shift+f12) to a field like mStatus in VSCode and it should show you any functions that use it, like this inline